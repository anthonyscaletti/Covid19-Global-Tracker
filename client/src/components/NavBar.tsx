import React from 'react';

import '../styles/navBar.css';
import Settings from './Settings';

class NavBar extends React.Component<IProps> {
    private renderLogo = () => {
        return (
            <div className="navLogo">
                <img src="../../BrandLogo1000x1000.png" alt="Logo"/>
            </div>
        );
    }

    private renderSettingsIcon = () => {
        return (
            <div className="navSettings" onClick={() => this.props.settingsClick()}>
                <img src="../../settingsIcon.png" alt="Settings"/>
            </div>
        );
    }

    render() {
        return (
            <div className="navBar">
                {this.renderLogo()}
                {this.renderSettingsIcon()}
            </div>
        );
    }
}

interface IProps {
    settingsClick: Function
}

export default NavBar;