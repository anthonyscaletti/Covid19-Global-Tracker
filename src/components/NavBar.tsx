import React from 'react'
import '../styles/navBar.css';

class NavBar extends React.PureComponent {
    render() {
        return (
            <div className="navBar">
                <div className="navLogo">
                    <img className="navLogoImg" src="../../BrandLogo1000x1000.png" alt="Logo"/>
                </div>
            </div>
        );
    }
}

export default NavBar;