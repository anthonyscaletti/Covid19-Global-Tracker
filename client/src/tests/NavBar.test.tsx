import React from 'react';
import renderer from 'react-test-renderer';

import NavBar from '../components/NavBar';

it('renders correctly', () => {
  const tree = renderer
    .create(<NavBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});