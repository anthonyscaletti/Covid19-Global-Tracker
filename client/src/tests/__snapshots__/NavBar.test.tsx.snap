// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navBar"
>
  <div
    className="navLogo"
  >
    <img
      alt="Logo"
      src="../../BrandLogo1000x1000.png"
    />
  </div>
  <div
    className="navSettings"
    onClick={[Function]}
  >
    <img
      alt="Settings"
      src="../../settingsIcon.png"
    />
  </div>
</div>
`;
